### PR TITLE
Matview context store BQ queries initial pass

### PR DESCRIPTION
**Pull request by pushyamig**
This is the PR for Bigquery version of queries for getting current term courses data. The queries here are not applied or used in BQ athletic views Current setup  since I have some question still waiting w.r.t how to check Matviews on a schedule run or if institution role is applied to Matviews as document [here](https://resources.unizin.org/display/UDP/UDP+Context+store) but for Postgres use case and not for BQ. 

1. I created 3 different files for 3 Matview queries for simplicity
2. There are syntax changes from BQ to Postgres. 
3. All these queries are taken from the document shared by Unizin and then I ran those queries from Big query console . Fixed any error w.r.t to sytax from that for example had to rename  `umich_materialized_views` to `udp-umich-prod.athletic_department_views` 
4. Take the BQ syntax queries from here run it in BQ console for error etc
5. Applied the Avg course grade changes later. I have these changes as individual commit to ease of PR review. The easy way to test if the course grade change are applied correctly is take the v1 query from `course_materialized_view.md` and run in the Postgres client and take the v1 query in BQ syntax run it from BQ console. compare with both query result for couple of courses if avg grade is equal. This approach is acceptable since we already validated Avg Course grade calculation with this PR #5 

Kyles Comments:
```
 [Here are the queries](https://docs.google.com/document/d/1Zqdkl5dyERWHMuedo0lc4zzObgwjd026EY2VSUU1Mho/edit?usp=sharing) that are scheduled to update in BQ for the materialized views.

 Note: The only changes I made were just formatting, and in the V2 view, I did a pre-sql block to query learner_activity_result first. The names of columns and tables and such are unchanged.
```

### CODE REVIEW

**Comment by pushyamig** [comment]
@zqian I have changed the queries extension from `.md` to '`.sql` to fit the SQL style syntax in VSCode, so it visually appealing. I have added SQL (BigQuery) VScode extension for this, but that is optional.

File: UDP/current_term_courses/current_term_courses_matview_bq.sql [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+   s.*,
+   c.academic_term_id,
+   c.credits
+ FROM
+   courses c
+ JOIN
+   sections s
+ ON
+   c.course_offering_id=s.course_offering_id),
+ enrollment AS (
```
Comment (zqian) [inline UDP/current_term_courses/current_term_courses_matview_bq.sql:61]: So this enrollment gets all enrollments from all courses in the database. Can we limit the scope to only current term courses/sections based on above courses_sections_of_current_term table?

Comment (pushyamig) [reply UDP/current_term_courses/current_term_courses_matview_bq.sql:61]: That's a reasonable request. I will work on it. I think event with this, the query ran very fast like in 1 mins.

Comment (pushyamig) [reply UDP/current_term_courses/current_term_courses_matview_bq.sql]: Done, I removed `enrollment` sub-set and did a join on `courses_sections_of_current_term` and `entity.course_section_enrollment` and then joined on `person` and `person_email`.  I checked with these courses (577751, 582106) which has students in multiple sections to see if it still preserve the student enrolled in multiple section and checking enrollment count.

Comment (zqian) [reply UDP/current_term_courses/current_term_courses_matview_bq.sql]: Looks good!

**Review by zqian — APPROVED** [review]

**Pull request merged**